c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -345,10 +459,62 @@ impl<'a, 'tcx> intravisit::Visitor<'tcx> for HirTyVisitor<'a, 'tcx> {
     }
 
     fn visit_item(&mut self, item: &'tcx hir::Item<'tcx>) {
+        let did = item.def_id.to_def_id();
         let field_ltys = &self.acx.gacx.field_ltys;
+
         #[allow(clippy::single_match)]
         match &item.kind {
-            ItemKind::Struct(VariantData::Struct(field_defs, _), _generics) => {
+            ItemKind::Struct(VariantData::Struct(field_defs, _), generics) => {
+                let adt_metadata = &self.acx.gacx.adt_metadata.table[&did];
+                let updated_lifetime_params = &adt_metadata.lifetime_params;
+
+                let original_lifetime_param_count = generics
+                    .params
+                    .iter()
+                    .filter(|p| matches!(p.kind, GenericParamKind::Lifetime { .. }))
+                    .count();
+
+                if updated_lifetime_params.len() != original_lifetime_param_count {
+                    let new_substs: Vec<_> = {
+                        let mut new_lifetime_params_iter = updated_lifetime_params.iter();
+
+                        let mut updated_lifetimes = vec![];
+                        let mut new_lifetimes = vec![];
+                        let mut other_params = vec![];
+
+                        for gp in generics.params {
+                            match gp.kind {
+                                GenericParamKind::Lifetime { .. } => {
+                                    let updated_lifetime_param = new_lifetime_params_iter
+                                        .next()
+                                        .expect("Not enough updated_lifetime_params");
+                                    updated_lifetimes.push(Rewrite::PrintTy(format!(
+                                        "{:?}",
+                                        updated_lifetime_param
+                                    )))
+                                }
+                                _ => {
+                                    other_params.push(Rewrite::PrintTy(gp.name.ident().to_string()))
```

Comment:
Will this swallow constraints, rewriting `struct Foo<T: Clone>(...)` into `struct Foo<T>(...)`?

Reply:
Issue created in https://github.com/immunant/c2rust/issues/927. Would you mind elaborating on that issue as to how fresh hypothetical lifetimes would be spliced in?